runtime/src/installed_scheduler_pool.rs

```diff
@@ -107,15 +145,261 @@ impl BankWithScheduler {
         }
     }
 
+    // take needless &mut only to communicate its semantic mutability to humans...
+    #[cfg(feature = "dev-context-only-utils")]
+    pub fn drop_scheduler(&mut self) {
+        self.inner.drop_scheduler();
+    }
+
+    pub(crate) fn wait_for_paused_scheduler(bank: &Bank, scheduler: &InstalledSchedulerRwLock) {
+        let maybe_result_with_timings = BankWithSchedulerInner::wait_for_scheduler(
+            bank,
+            scheduler,
+            WaitReason::PausedForRecentBlockhash,
+        );
+        assert!(
+            maybe_result_with_timings.is_none(),
+            "Premature result was returned from scheduler after paused"
+        );
+    }
+
+    #[must_use]
+    pub fn wait_for_completed_scheduler(&self) -> Option<ResultWithTimings> {
+        BankWithSchedulerInner::wait_for_scheduler(
+            &self.inner.bank,
+            &self.inner.scheduler,
+            WaitReason::TerminatedToFreeze,
+        )
+    }
+
     pub const fn no_scheduler_available() -> InstalledSchedulerRwLock {
         RwLock::new(None)
     }
 }
 
+impl BankWithSchedulerInner {
+    #[must_use]
+    fn wait_for_completed_scheduler_from_drop(&self) -> Option<ResultWithTimings> {
+        Self::wait_for_scheduler(
+            &self.bank,
+            &self.scheduler,
+            WaitReason::DroppedFromBankForks,
+        )
+    }
+
+    #[must_use]
+    fn wait_for_scheduler(
```

Comment:
I think the top-level `wait_for_X` fns make sense naming wise, but this one I think is a bit confusing. The other naming, for example `wait_for_completed_scheduler` will wait until we have a completed scheduler; this one doesn't have the same convention since we aren't waiting to have a scheduler. or maybe I am misunderstanding the use and confused myself.

Reply:
how about this?: a1145ea80b4b1d08d2773de866f1829985c0fb70

Reply:
yeah, perfect!